doc/user-guide.md

```diff
@@ -423,6 +423,88 @@ To implement complex deletion logic, you can add a finalizer to your Custom Reso
 deleted until you remove the finalizer (ie, after your cleanup logic has successfully run). For more information, see the
 [official Kubernetes documentation on finalizers](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#finalizers).
 
+**Example:**
+
+After the metadata name line your CR file add the reference to `finalizers`. If you are following the above example your CR file should look something like this:
+
+```sh
+$ cat deploy/crds/cache_v1alpha1_memcached_cr.yaml
+apiVersion: cache.example.com/v1alpha1
+kind: Memcached
+metadata:
+  name: example-memcached
+  finalizers:
+  - finalizer.cache.example.com
+spec:
+  # Add fields here
+  size: 3
+```
+
+Following is a snippet from the controller file under `pkg/controller/memcached/memcached_controller.go`
+
+```go
+func (r *ReconcileMemcached) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+    reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+    reqLogger.Info("Reconciling Memcached")
+        
+    // Fetch the Memcached instance
+    memcached := &cachev1alpha1.Memcached{}
+    err := r.client.Get(context.TODO(), request.NamespacedName, memcached)
+
+    //{...} reconcile implementations ...
+    
+    //Perform the actions required to allow the CR deletion
+    if err := r.doFinalizerActions(memcached); err != nil {
+        reqLogger.Error(err, "Unable to perform actions to finalize the Mencached", "Memcached.Namespace", memcached.Namespace, "Memcached.Name", memcached.Name)
+        return reconcile.Result{}, err
+    }
+    
+    //Check Finalizer
+    if err := r.checkFinalizer(memcached); err != nil {
+        reqLogger.Error(err, "Unable to satisfy finalizer conditionals for Mencached", "Memcached.Namespace", memcached.Namespace, "Memcached.Name", memcached.Name)
+        return reconcile.Result{}, err
+    }
+    
+    return reconcile.Result{}, nil
+}
+
+//checkFinalizer returns error when was not possible check that the defined actions were performed with success
+func (r *ReconcileMemcached) checkFinalizer(m *cachev1alpha1.Memcached) error {
+	if len(m.GetFinalizers()) > 0 && m.GetDeletionTimestamp() != nil {
+		// Check if finalizer conditional was performed with sucess
+		if  isFinalizerActionMet(m) {
+			return fmt.Errorf("Unable to met criteria for Mancached deletion.")
+		}
+		
+		// Remove finalizer and deletion timestamp from the CR to allow the cluster to delete it
+		m.SetFinalizers(nil)
+		m.SetDeletionTimestamp(nil)
```

Comment:
I don't think that we should be clearing the deletion timestamp, this is changing what the user asked for.

Reply:
Thank you for let me know.